test/functional/system/rounds.js

```diff
@@ -0,0 +1,1691 @@
+					block.reward
+				);
+			} else {
+				rewards[publicKey] = {
+					publicKey,
+					fees: new Bignum(feesPerDelegate),
+					rewards: new Bignum(block.reward),
+				};
+			}
+
+			if (index === blocks.length - 1) {
+				// Apply remaining fees to last delegate
+				rewards[publicKey].fees = rewards[publicKey].fees.plus(feesRemaining);
+			}
+		});
+
+		_.each(rewards, delegate => {
+			delegate.fees = delegate.fees.toString();
+			delegate.rewards = delegate.rewards.toString();
+		});
+
+		return rewards;
+	}
+
+	function addTransaction(transaction, cb) {
+		__testContext.debug(`	Add transaction ID: ${transaction.id}`);
+		// Add transaction to transactions pool - we use shortcut here to bypass transport module, but logic is the same
+		// See: modules.transport.__private.receiveTransaction
+		transaction = library.logic.transaction.objectNormalize(transaction);
+		library.balancesSequence.add(sequenceCb => {
+			library.modules.transactions.processUnconfirmedTransaction(
+				transaction,
+				true,
+				err => {
+					if (err) {
+						return setImmediate(sequenceCb, err.toString());
+					}
+					return setImmediate(sequenceCb, null, transaction.id);
+				}
+			);
+		}, cb);
+	}
+
+	function getNextForger(offset, cb) {
+		offset = !offset ? 1 : offset;
+		const lastBlock = library.modules.blocks.lastBlock.get();
+		const slot = slots.getSlotNumber(lastBlock.timestamp);
+		library.modules.delegates.generateDelegateList(
+			lastBlock.height + 1,
+			null,
+			(err, delegateList) => {
+				if (err) {
+					return cb(err);
+				}
+				const nextForger = delegateList[(slot + offset) % slots.delegates];
+				return cb(nextForger);
+			}
+		);
+	}
+
+	function forge(cb) {
+		const transactionPool = library.rewiredModules.transactions.__get__(
+			'__private.transactionPool'
+		);
+
+		async.waterfall(
+			[
+				transactionPool.fillPool,
+				function(waterCb) {
+					getNextForger(null, delegatePublicKey => {
+						waterCb(null, delegatePublicKey);
+					});
+				},
+				function(delegate, waterCb) {
+					let lastBlock = library.modules.blocks.lastBlock.get();
+					const slot = slots.getSlotNumber(lastBlock.timestamp) + 1;
+					const keypair = keypairs[delegate];
+					__testContext.debug(
+						`		Last block height: ${lastBlock.height} Last block ID: ${
+							lastBlock.id
+						} Last block timestamp: ${
+							lastBlock.timestamp
+						} Next slot: ${slot} Next delegate PK: ${delegate} Next block timestamp: ${slots.getSlotTime(
+							slot
+						)}`
+					);
+					library.modules.blocks.process.generateBlock(
+						keypair,
+						slots.getSlotTime(slot),
+						err => {
+							if (err) {
+								return waterCb(err);
+							}
+							lastBlock = library.modules.blocks.lastBlock.get();
+							__testContext.debug(
+								`		New last block height: ${
+									lastBlock.height
+								} New last block ID: ${lastBlock.id}`
+							);
+							return waterCb(err);
+						}
+					);
+				},
+			],
+			cb
+		);
+	}
```

Comment:
There is already a method `require(./common).forge` which have same code. We should use it, or update it some relevant change is required.